### DISCUSSION TITLE
How do I accept just one word from `copilot`/`copilot-cmp`?

### DISCUSSION THREAD

**Discussion by Zeta611** [LazyVim/LazyVim#2109] [Q&A] [answered]
Most of the times, I just want a single word or two from the suggestions from the Copilot. However, `<cr>` from the `copilot-cmp`/`nvim-cmp` fills in the whole suggestion.
How can I customize this behavior?

I enabled `coding.copilot` from LazyVim Extras.

**Comment by dpetka2001** [comment]
`copilot.lua` seems to have a [mapping](https://github.com/zbirenbaum/copilot.lua#setup-and-configuration) about `accept_word` that seems to be disabled by default. Unfortunately, I don't use Copilot myself, so can't test this for you. Maybe you'd like to check that out.

**Reply by dpetka2001** [reply]
It seems that this setting is interfering with `copilot-cmp` and is suggested to be disabled. And from a few issues at `nvim-cmp` repo [here](https://github.com/hrsh7th/nvim-cmp/issues/1536) and [here](https://github.com/zbirenbaum/copilot-cmp/issues/63) seems like the maintainer is not eager to support this directly. So, your best option would be to enable this for yourself and probably disable `cmp-copilot` since it might cause conflicts? Not 100% sure, since I don't use it myself as I've already mentioned.

**Reply by Zeta611** [reply]
I tried disabling `copilot-cmp`:
```lua
return {
  "zbirenbaum/copilot-cmp",
  enabled = false,
}
```
but now copilot doesn't start automatically, and even if I start it via `:LspStart Copilot`, it doesn't recommend a completion. I thought it should appear as a virtual text?

**Reply by dpetka2001** [reply]
By default LazyVim disables suggestions by `copilot.lua` as you can see [here](https://www.lazyvim.org/extras/coding/copilot#copilotlua) (because that is suggested by `copilot-cmp`). You should enable the suggestion option in your own config.

**Reply by Zeta611** [reply]
Thanks for the pointer. I'm still struggling to make it work with the following :(
```lua
return {
  "zbirenbaum/copilot.lua",
  opts = {
    suggestion = {
      enabled = true,
    },
    keys = {
      {
        "<C-l>",
        require("copilot.suggestion").accept_word,
      },
    },
  },
}
```

**Reply by dpetka2001** [reply]
How about
```lua
suggestion = {
	enabled = true,
	keymap = {
		accept_word = "<C-l>",
	},
},
```

**Reply by Zeta611** [reply, marked as answer]
Thank you! I ended up disabling `coding.copilot` extra---so `enabled` is by default `true` now---and used the following options:
```lua
return {
  "zbirenbaum/copilot.lua",
  opts = {
    suggestion = {
      auto_trigger = true,
      keymap = {
        accept_word = "<C-l>",
      },
    },
    filetypes = {
      ["."] = true,
    },
  },
}
```
I guess `auto_trigger = true` part is also important, as it needs to work along side the `nvim-cmp` package.

**Reply by dpetka2001** [reply]
Cool glad you got it working.
